### PR TITLE
fix: `Build Sample App Tests` and `integ_next_auth_authenticator_and_ssr_page` tests fix

### PR DESCRIPTION
**Pull request by kvramyasri7**
#### Description of changes

- Dropping `@angular/cli` version to `16.2.10`. In attempt of fixing  [Build Sample App Tests](https://github.com/aws-amplify/amplify-js/actions/runs/7035198896/job/19145030133)
- Limit `integ_next_auth_authenticator_and_ssr_page` to edge & chrome to align with [e2e tests](https://github.com/aws-amplify/amplify-js/blob/main/.github/integ-config/integ-all.yml#L501)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/canary-config/canary-all.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@ minimal_browser_list: &minimal_browser_list
   - chrome
   - firefox
 
+minimal_browser_list_2: &minimal_browser_list_2
```
Comment (elorzafe) [inline .github/canary-config/canary-all.yml:5]: The name is not very good

Why not rename browser list to `small`, `medium`, `large`?

wdyt?

Comment (kvramyasri7) [reply .github/canary-config/canary-all.yml:5]: OOOOOhh I though `minimal_small` and `minimal_medium` and `minimum_ large` does not make Any sort of sense is understanding Francisco Rodriguez @elorzafe

File: .github/canary-config/canary-all.yml [hunk 2]
```diff
@@ -305,4 +309,4 @@ tests:
     category: auth
     sample_name: [auth-ssr]
     spec: auth-ssr
-    browser: *minimal_browser_list
+    browser: *minimal_browser_list_2
```
[the remark this answers is not in the document]
Comment (ashwinkumar6) [reply .github/canary-config/canary-all.yml:312]: I was thinking along the lines of canary being more like a subset of e2e. I think it makes sense to run this on edge as well if required. Maybe we can run this test on edge on e2e and canary. wdyt 

Also since `minimal_browser_list_2` was only used by this test for now I was wondering if we could just specify `browser: [chrome, edge]` instead of creating the new variable. I don't have a strong opinion though

File: .github/canary-config/canary-all.yml (hunk 1 above)
Comment (elorzafe) [reply .github/canary-config/canary-all.yml:5]: I was thinking something more like 
- `browser_list_small`
- `browser_list_medium`
- `browser_list_large`
`minimal_browser_list_2` means there is another `minimal_browser_list` but doesn't give me any information related to that parameter.

If names doesn't make sense I would prefer to list the values inline. The idea is to understand whats going in a simple manner.

File: .github/canary-config/canary-all.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashwinkumar6) [reply .github/canary-config/canary-all.yml:312]: I think we mostly try to use `minimal_browser_list` in all places. Except some special tests which only run on specific browsers instead of the entire `minimal_browser_list`. For this we add the browser list inline
(This is something we currently do for e2e) 

I think adding browser list inline might also help identify these special tests in future and troubleshoot why it doesn't work for a particular browser.  So i think it's fine if it's slightly inconsistent

[the remark this answers is not in the document]
Comment (kvramyasri7) [reply .github/canary-config/canary-all.yml]: @ashwinkumar6  can you takē a l00k at it now please 🙏🏼
